python/ray/serve/scripts.py

```diff
@@ -159,19 +159,34 @@ def cli():
     default=DeploymentMode.HeadOnly,
     required=False,
     type=click.Choice(list(DeploymentMode)),
-    help="Location of the HTTP proxies. Defaults to HeadOnly.",
+    help="DEPRECATED: Use `--proxy-location` instead.",
 )
-def start(address, http_host, http_port, http_location):
+@click.option(
+    "--proxy-location",
+    default=DeploymentMode.EveryNode,
+    required=False,
+    type=click.Choice(list(DeploymentMode)),
+    help="Location of the HTTP proxies. Defaults to EveryNode.",
+)
+def start(address, http_host, http_port, http_location, proxy_location):
+    if http_location != DeploymentMode.HeadOnly:
+        cli_logger.warning(
+            "The `--http-location` flag to `serve start` is deprecated, "
+            "use `--proxy-location` instead."
+        )
+
+        proxy_location = http_location
```

Comment:
If both proxy location and http location are specified, should we have proxy location take priority?

Reply:
hmm we should, but as written this is not straightforward to detect because our defaults aren't "None"

Reply:
I'm not too worried about this, using both would be very strange/accidental and if users don't get the behavior they expect it'll be pretty clear why